File: project/jimmer-core/src/test/java/org/babyfish/jimmer/LongDataTest.java
package org.babyfish.jimmer;

import com.fasterxml.jackson.core.JsonProcessingException;
import org.babyfish.jimmer.model.LongData;
import org.babyfish.jimmer.model.LongDataDraft;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

public class LongDataTest {

    @Test
    public void test() throws JsonProcessingException {
        LongData data = LongDataDraft.$.produce(draft -> {
            draft.setNotNullValue(1L);
            draft.setNullableValue(2L);
            draft.setValues(Arrays.asList(3L, 4L));
        });
        String json = data.toString();
        Assertions.assertEquals(
                "{\"notNullValue\":\"1\",\"nullableValue\":\"2\",\"values\":[\"3\",\"4\"]}",
                json
        );
        LongData data2 = ImmutableObjects.fromString(LongData.class, json);
        Assertions.assertEquals(
                data,
                data2
        );
    }
}

File: project/jimmer-core-kotlin/src/test/kotlin/org/babyfish/jimmer/kt/LongDataTest.kt
package org.babyfish.jimmer.kt

import org.babyfish.jimmer.ImmutableObjects
import org.babyfish.jimmer.kt.model.LongData
import org.babyfish.jimmer.kt.model.by
import kotlin.test.Test
import kotlin.test.expect

class LongDataTest {

    @Test
    fun test() {
        val data = new(LongData::class).by {
            nonNullValue = 1L
            nullableValue = 2L
            values = listOf(1L, 2L)
        }
        val json = data.toString()
        expect(
            """{"nonNullValue":"1","nullableValue":"2","values":["1","2"]}"""
        ) {
            json
        }
        val data2 = ImmutableObjects.fromString(LongData::class.java, json)
        expect(data) {
            data2
        }
    }
}